feeder/src/main/java/datawave/microservice/feeder/file/FeederFileScanner.java

```diff
@@ -52,28 +45,19 @@ protected void process(Path workingFile) throws IOException {
             }
         }
         
-        log.info("Moving file: " + workingFile + " to: " + fileTarget);
+        log.info("Moving file: {} to: {}", workingFile, fileTarget);
         FileSystem fs = FileSystem.get(workingFile.toUri(), conf);
         Path parentDir = fileTarget.getParent();
-        if (!fs.exists(parentDir)) {
-            if (!fs.mkdirs(parentDir)) {
-                log.warn("Failed to create " + parentDir);
-                throw new IOException("Failed to create " + parentDir);
-            }
+        if (!fs.exists(parentDir) && !fs.mkdirs(parentDir)) {
+            log.warn("Failed to create {}", parentDir);
+            throw new IOException("Failed to create " + parentDir);
         }
         
         if (fs.rename(workingFile, fileTarget)) {
-            while (!feedSource.send(MessageBuilder.withPayload(fileTarget + "," + feederProperties.getInputFormatClass() + "," + feederProperties.getDataType())
-                            .build())) {
-                try {
-                    // TODO configurable?
-                    Thread.sleep(5000);
-                } catch (InterruptedException e) {
-                    // no-op
-                }
-            }
+            feedSource.send(MessageBuilder.withPayload(fileTarget + "," + feederProperties.getInputFormatClass() + "," + feederProperties.getDataType())
+                            .build());
```

Comment:
should there be some sort of "if failed, log something" added for this?

Reply:
sure. Easy to add